hometask1/foo.rb

```diff
@@ -0,0 +1,46 @@
+class Foo
+  @@counter = 0
+  @instance_counter = 0
```

Comment:
You don't use it. **At all**.
In next lines, you are managing another `@instance_counter` which doesn't belong to `Foo` as a class-object (like in this context), but to concrete instance of `Foo`.